src/v/redpanda/application.cc

```diff
@@ -334,6 +336,15 @@ void application::initialize(
           config::shard_local_cfg().zstd_decompress_workspace_bytes());
     }).get0();
 
+    /*
+     * allocate per-core workspaces for async_stream_zstd. it can be several
+     * megabytes in size, so do it before memory becomes fragmented.
+     */
+    ss::smp::invoke_on_all([] {
```

Comment:
Might as well just move this inside the `invoke_on_all` above, no? Seems equally clean and we don't need to duplicate the cost of the cross-shard invoke.

Reply:
Moving both into the single `invoke_on_all`